rfcs/convergence/authoring-stories.md

```diff
@@ -433,6 +434,48 @@ export const Default = (props: PopoverProps) => (
 Public stories should only contain code, which is useful for users to see after clicking on “Show code” in documentation.
 Extra markup (e.g., container with CSS styles) can be added via [Decorators](https://storybook.js.org/docs/react/writing-stories/decorators).
 
+### 10. Internal stories for testing
+
+As mentioned in for E2E testing, we should ensure maximum coverage for all publicly viewable stories by our consumers.
+For more complex scenarios that need to be tested we should make sure that stories exist for E2E tests but should not be
+easily accessible publicly.
+
+Storybook has proposed a feature for this in [storybookjs/storybook#9209](https://github.com/storybookjs/storybook/issues/9209)
+which will configure stories to exist in deeplink URL format, but do not appear in the nav tree or the docs page. As stated in the issue,
+we can workaround before the release of this feature by modifying `manager-head.html` and set `display:none`for all
+stories with a specific DOM `id` attribute. Storybook uses the `id` attribute for each link in the nav tree, and sets
+the value to the story id.
+
+We propose to use an extra filename extension and naming convention for internal stories:
+
+```ts
+// MenuTabstops.internal.stories.tsx
+// Deep link /story/components-menu--tabstops-internal
+// Does not appear in the sidebar or docs page
+export const TabstopsInternal = () => {
+  // story
+};
+```
+
+The filename extension `.internal.` is used for IDE searchability and codebase readability.
```

Comment:
There is no way to enforce an extra `.internal.` extension, but it does benefit searchability, should this just be removed ?

Reply:
Thinking about file naming, I realized that right now, these three can be completely different:
1. file name
2. name of export
3. displayed story name

I think this could get messy. Perhaps it’d be good to require some consistency between file name and name of export. 

We already require that file name starts with component name. We could add a requirement that export name is either
- the same as file name
- or same as file name without the component name prefix

Do you think this would be an improvement? 
This would then require internal stories to end file names with `Internal.stories.tsx`, which will work well with search too.

Reply:
yup sounds good